### ISSUE TITLE
[ENHANCEMENT] Performance impact when getting playlist with high videos count 

### ISSUE THREAD

**Issue by SteeledSlagle13** [VincentJonathan/scrape-yt#12]
**Describe the bug**
Hey so I'm seeing when I call
```JS
scrapeYt.getPlaylist(playlistMatch[1]).then(playlist => {
    const playlistVideos: Video[] = playlist['videos'];

   console.log(playlistVideos);
});
```
Where playlistMatch[1] = `PLqFSI8ggBE67YFjGyLykEqeh0c37uC8fi`

My audio playing on discord has a gap when this is called

**To Reproduce**
- should be able to reproduce on any playlist over 30+ if not 30+ then try 100 or more
- lower numbered playlist dont have this issue as prevalent as when it is a bigger playlist

**Expected behavior**
No gap in audio when calling this method

**Screenshots**
N/A

**Additional context**
- this is with a discord bot
- seeing in all 3 of my environments (mac - local, widows - local, ubuntu - aws)
- im using typescript

sorry for opening so many issue man :/

i tried separating the call into a diff method that isnt connected to the player and calling the getPlaylist function while the something is playing and got the same issue.... I dont see the issue with getVideo() and im not seeing it with the search as well

**Comment by SteeledSlagle13** [comment]
if u have discord i can invite u to my test server to show u

**Comment by SuspiciousLookingOwl** [comment]
Can you share your minimum code for discord bot with the command to play song and get playlist? (Don't forget to remove any sensitive data like discord token)

**Comment by SteeledSlagle13** [comment]
ye :D ill get that up here in a bit

**Comment by SteeledSlagle13** [comment]
are u on windows?

**Comment by SuspiciousLookingOwl** [comment]
Yes, but i also have VPS running on Ubuntu, if i can't reproduce it in my system, i will try to host it on my VPS.

**Comment by SteeledSlagle13** [comment]
okie dokie ill remove the part with ffmpeg location instead of worrying about that

**Comment by SteeledSlagle13** [comment]
hey u cool if i email it to u?

**Comment by SuspiciousLookingOwl** [comment]
Sure,  [me@vincentjonathan.com](mailto:me@vincentjonathan.com)

**Comment by SteeledSlagle13** [comment]
just sent email with code and steps lmk if you have questions!!!

**Comment by SuspiciousLookingOwl** [comment]
Try update to 1.2.5 (alpha), `npm i scrape-yt@1.2.5` and try again.

**Comment by SteeledSlagle13** [comment]
okie dokie give me like two minutes XD

**Comment by SuspiciousLookingOwl** [comment]
Oh nvm it didn't fixed it lol, i thought it did! I will take a look again on what causes it.

**Comment by SteeledSlagle13** [comment]
XD hahahah sounds good im on 1.2.5 now and no dice I'll keep checking back here
i appreciate you!

**Comment by SuspiciousLookingOwl** [comment]
This seems like a performance issue from the scraper (cheerio). I'm still looking for a lighter alternative to try.

**Comment by SteeledSlagle13** [comment]
oof sounds good im not in a hurry for the fix i just wanted to bring it up
thank you so much for looking into it i really appreciate it

**Comment by SuspiciousLookingOwl** [comment]
Np! One easy temporary solution that i can think of is just make a local HTTP server as an API to get youtube data using my library, so instead of using `scrape-yt` directly on the discord code, you just send a HTTP request to that local server so the scraper isn't hogging the discord node js thread.

**Comment by SteeledSlagle13** [comment]
oh thats an idea
i can look into that :D

**Comment by SteeledSlagle13** [comment]
hey just to let you know.... i tried the node server... that works very nicely!!!! thank you very much

**Comment by SuspiciousLookingOwl** [comment]
I've added worker thread support which allows the lib to run the scraper in parallel. Tried it on my VPS and seems like it fixed the issue. It's not really a performance improvement, but still 😬. Update not yet published on npm.

**Comment by SuspiciousLookingOwl** [comment]
Published on npm (@1.3.2). You just need to add options on the second parameter and add `useWorkerThread: true`

```js
import scrapeYt from "scrape-yt";
// Changed from -> import { scrapeYt } from "scrape-yt";

scrapeYt.getPlaylist("PLx65qkgCWNJIgVrndMrhsedBz1VDp0kfm", { useWorkerThread: true }).then(playlist => {
  console.log(playlist);
});
```

**Comment by SteeledSlagle13** [comment]
dang thats awesome!!!! thank you so much
ill get this implemented this week 😀
